external-crates/move/move-compiler/src/hlir/translate.rs

```diff
@@ -1412,6 +1446,23 @@ fn exp_impl(
     H::exp(ty, sp(eloc, res))
 }
 
+fn struct_name(sp!(_, t): &H::Type) -> Option<StructName> {
+    let H::Type_::Single(st) = t else {
+        return None;
+    };
+    let bt = match &st.value {
+        H::SingleType_::Base(bt) => bt,
+        H::SingleType_::Ref(_, bt) => bt,
+    };
+    let H::BaseType_::Apply(_, tname ,_ ) = &bt.value else {
+        return None;
+    };
+    if let H::TypeName_::ModuleType(_, struct_name) = tname.value {
+        return Some(struct_name);
```

Comment:
Do we need to check for the current module here? The error might be a bit flaky otherwise? 

At the very least let's add the following test
```
module a::other {
    struct X { f: u64 }
}
module a::m {
    struct X { f: u64 }
    public fun flaky(x: a::other::X): u64 { x.f }
}
```
If "private" field check is NonBlocking, then we will need to check the module. 

If we do not currently need to check the module name because "private" fields are a BlockingError, then this is all moot, but the test will make sure that we update this code if we change that.

Reply:
I had this check in the earlier version implemented with typed AST but with hlir we don't have access to current module in context (though it's surely not hard to add). In the meantime, I added the suggested test.